fights/envs/puoribor.py

```diff
@@ -152,7 +152,16 @@ class PuoriborEnv(BaseEnv):
     """
 
     def step(
-        self, state: PuoriborState, agent_id: int, action: ArrayLike
+        self,
+        state: PuoriborState,
+        agent_id: int,
+        action: ArrayLike,
+        pre_callback: Union[
+            Callable[[PuoriborState, int, ArrayLike], None], None
+        ] = None,
+        post_callback: Union[
+            Callable[[PuoriborState, int, ArrayLike], None], None
+        ] = None,
```

Comment:
`Optional[...]` should be preferred over `Union[..., None]`.
Also, let's make `pre_callback` and `post_callback` [keyword only arguments](https://peps.python.org/pep-3102/).

Reply:
Is it just me, or does `pre_callback` sound a bit like a function that runs after `callback`?
Maybe we should call it `pre_step_fn` or  `pre_step_callback`  or `callback_pre` or something similar?